storage-node-v2/src/commands/server.ts

```diff
@@ -29,34 +40,177 @@ export default class Server extends ApiCommandBase {
       required: true,
       description: 'Server port.',
     }),
+    sync: flags.boolean({
+      char: 's',
+      description: 'Enable data synchronization.',
+      default: false,
+    }),
+    syncInterval: flags.integer({
+      char: 'i',
+      description: 'Interval between synchronizations (in minutes)',
+      default: 1,
+    }),
+    queryNodeHost: flags.string({
+      char: 'q',
+      required: true,
+      description: 'Query node host and port (e.g.: some.com:8081)',
+    }),
+    syncWorkersNumber: flags.integer({
+      char: 'r',
+      required: false,
+      description: 'Sync workers number (max async operations in progress).',
+      default: 20,
+    }),
+    elasticSearchHost: flags.string({
+      char: 'e',
+      required: false,
+      description: `Elasticsearch host and port (e.g.: some.com:8081).
+Log level could be set using the ELASTIC_LOG_LEVEL enviroment variable.
+Supported values: warn, error, debug, info. Default:debug`,
+    }),
+    disableUploadAuth: flags.boolean({
+      char: 'a',
+      description: 'Disable uploading authentication (should be used in testing-context only).',
+      default: false,
+    }),
     ...ApiCommandBase.flags,
   }
 
   async run(): Promise<void> {
     const { flags } = this.parse(Server)
 
+    await removeTempDirectory(flags.uploads, TempDirName)
+
+    let elasticUrl
+    if (!_.isEmpty(flags.elasticSearchHost)) {
+      elasticUrl = `http://${flags.elasticSearchHost}`
+      initElasticLogger(elasticUrl)
+    }
+
+    const queryNodeUrl = `http://${flags.queryNodeHost}/graphql`
+    logger.info(`Query node endpoint set: ${queryNodeUrl}`)
+
     if (flags.dev) {
       await this.ensureDevelopmentChain()
     }
 
+    if (flags.disableUploadAuth) {
+      logger.warn(`Uploading auth-schema disabled.`)
+    }
+
+    if (flags.sync) {
+      logger.info(`Synchronization enabled.`)
+
+      runSyncWithInterval(flags.worker, queryNodeUrl, flags.uploads, flags.syncWorkersNumber, flags.syncInterval)
+    }
+
     const account = this.getAccount(flags)
     const api = await this.getApi()
 
+    await verifyWorkerId(api, flags.worker, account)
+
     try {
       const port = flags.port
       const workerId = flags.worker ?? 0
```

Comment:
This is redundant because `flags.worker` is required:
```suggestion
      const workerId = flags.worker
```

Reply:
Fixed here: https://github.com/Joystream/joystream/pull/2686/commits/7230f23e8e30966c0cac51a17f528926f6069d0b